Confidence/src/main/java/com/spotify/confidence/DebugLogger.kt

```diff
@@ -0,0 +1,35 @@
+package com.spotify.confidence
+
+import android.util.Log
+
+internal class DebugLogger {
+    var level: DebugLoggerLevel = DebugLoggerLevel.NONE
+
+    internal fun logEvent(tag: String, event: EngineEvent, details: String) {
+        when (level) {
+            DebugLoggerLevel.VERBOSE -> Log.v(tag, details + event.toString())
+            DebugLoggerLevel.DEBUG -> Log.d(tag, details + event.eventDefinition)
+            DebugLoggerLevel.NONE -> {
+                // do nothing
+            }
+        }
+    }
+
+    internal fun logMessage(tag: String, message: String, isWarning: Boolean = false) {
+        if (!isWarning) {
+            when (level) {
+                DebugLoggerLevel.VERBOSE -> Log.v(tag, message)
```

Comment:
these cases seem all to be the same. could you explain the logic here please? what is `isWarning` and why that matters?

Reply:
I think we can combine Verbose and Debug, since they both can handle logging exceptions. A warning, from what I understand, would show up differently in the Logcat, so I wanted to handle it differently - and right now you'd only get a warning if the storage is full.